samcli/commands/local/lib/local_lambda.py

```diff
@@ -72,7 +72,8 @@ def invoke(self, function_name, event, stdout=None, stderr=None):
         config = self._get_invoke_config(function)
 
         # Invoke the function
-        self.local_runtime.invoke(config, event, debug_context=self.debug_context, stdout=stdout, stderr=stderr)
+        with function.codeuri:
```

Comment:
So the way this works is the `function.codeuri` is not set to a class which is a context manager?

Reply:
yes, it is like that. I did this to preserve most of the existing code